back-end/API/app/services/user_service.py

```
@@ -44,7 +44,15 @@ def register(self, data):
         if not current_app.config.get('TESTING'):
             if not captcha_service.verify_captcha(data['captchaToken']):
                 return jsonify({'message': 'Captcha verification failed'}), 400
-        return auth_repo.register(data)
+        try:
+            new_user = auth_repo.register(data) 
+            token = self._generateToken(new_user)  
+            return token,new_user 
```

Ruff (0.15.7)

[warning] Consider moving this statement to an `else` block

(TRY300)